app/src/services/mongo/MongoDBConnection.ts

```diff
@@ -0,0 +1,79 @@
+import mongoose, {Mongoose} from 'mongoose';
+import ApplicationException from '../../exceptions/ApplicationException';
+import {getLogger} from '../../logger';
+import {injectable} from 'inversify';
+import Database, {DatabaseConfig} from '../../config/database';
+
+@injectable()
+export default class MongoDBConnection {
+    private static readonly logger = getLogger('DatabaseConnection');
+    private static isConnected = false;
+    private static db: Mongoose;
+    private static dbConfig: Database;
+
+    constructor(dbConfig: Database) {
+        MongoDBConnection.dbConfig = dbConfig;
+    }
+
+    public getConnection(): Mongoose {
+        if (MongoDBConnection.isConnected) {
+            return MongoDBConnection.db;
+        } else {
+            try {
+                void MongoDBConnection.connect();
+                return MongoDBConnection.db;
+            } catch (error) {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,@typescript-eslint/restrict-template-expressions
+                throw new ApplicationException(`CRITICAL: UNABLE TO OPEN DATABASE CONNECTION! ${error.message ?? error}`, 'database/mongo-connection', 1);
+            }
+        }
+    }
+
+    public terminateConnection(): boolean {
+        if (!MongoDBConnection.isConnected) {
+            MongoDBConnection.logger.error('terminateConnection was called when not connected!');
+            return true;
+        }
+
+        try {
+            void MongoDBConnection.db.connection.close();
+        } catch (error) {
+            MongoDBConnection.logger.emerg('Was unable to close the database connection! FUBAR');
+            throw new ApplicationException('CRITICAL: UNABLE TO CLOSE DATABASE CONNECTION! FUBAR!', 'database/mongo-connection', 1);
+        }
+
+        return true;
+    }
+
+    private static async connect(): Promise<void> {
+        MongoDBConnection.logger.info('Starting connection...');
+        const config: DatabaseConfig = MongoDBConnection.dbConfig.config;
+
+        const connStr = `mongodb://${config.user}:${config.pass}@${config.host}:${config.port}?authSource=admin`;
+        MongoDBConnection.logger.debug(connStr);
+
+        mongoose.connection.on('connected', function() {
```

Comment:
Don't know if the `connection` property is available before the you call `connect`, but the listener should only be added once so maybe add a connection prepare method that is called on the constructor or on initialization.

Also maybe add some of the other listeners that log at different levels(for debugging mostly).

Reply:
I think connection events should always be available, it's an event emitter and it's just merely being registered. Think it's ok here.

Reply:
As long as they are only added once, should be fine